### PR TITLE
Migrate `prow` component deployment to `flux`

### PR DESCRIPTION
**Pull request by oliver-goetz**
**What this PR does / why we need it**:

Migrates the `prow` component deployment from `deploy.sh` to flux.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener-community/hackathon/tree/main/2024-12_Schelklingen
Follow-up to https://github.com/gardener/ci-infra/pull/2812

**Special notes for your reviewer**:

### CODE REVIEW

**Review by timebertt — COMMENTED** [review]
Awesome!
/lgtm
Left one idea :)

File: deploy/prow/prowjob_customresourcedefinition.yaml
[the remark this answers is not in the document]
Comment (oliver-goetz) [reply deploy/prow/prowjob_customresourcedefinition.yaml:1]: Good idea 👍

**Review by timebertt — APPROVED** [review]

**Pull request merged**